### PR TITLE
feat: Add drag and drop support in event type list

### LINKED ISSUE #10016: Allow drag and drop to rearrange event types
### Is your proposal related to a problem?

It would be nice if we could drag and drop the event types instead of using arrows to move them one step up or down.

https://github.com/calcom/cal.com/assets/47350983/71919323-eeba-4828-8ec5-1e95414f35b8

### Describe the solution you'd like

Implement drag and drop using [react-beautiful-dnd](https://www.npmjs.com/package/react-beautiful-dnd?activeTab=readme).

### Describe alternatives you've considered

There are other libraries like react-dnd but react-beautiful-dnd seems to be the most popular one.

### Additional context

There was an issue that was raised before and was closed as it was low priority and since drag and drop tends to be flaky: https://github.com/calcom/cal.com/issues/7125.
With this in mind this feature can be considered a prototype, which will need extensive testing before it can be submitted.

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

Allow Drag and drop support
Fixes #10016 
/claim #10016

https://github.com/calcom/cal.com/assets/81948346/7cc0ca2c-ab32-40d8-b57d-3e332123c9fd

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by O4epegb** [comment]
react-beautiful-dnd is basically on life support, and did't have any new features for around 3 years now, maybe it's worth consider something more modern and maintained

File: package.json [hunk 1]
```diff
@@ -92,9 +92,11 @@
     "vitest-mock-extended": "^1.1.3"
   },
   "dependencies": {
+    "@types/react-beautiful-dnd": "^13.1.4",
     "city-timezones": "^1.2.1",
     "eslint": "^8.34.0",
     "lucide-react": "^0.171.0",
+    "react-beautiful-dnd": "^13.1.1",
```
Comment (O4epegb) [inline package.json:99]: react-beautiful-dnd is basically on life support, and did't have any new features for around 3 years now, maybe it's worth consider something more modern and maintained

**Comment by O4epegb** [comment]
It seems like another person already made a PR before https://github.com/calcom/cal.com/pull/10017?

**Comment by anikdhabal** [comment]
> It seems like another person already made a PR before #10017?

The PR was in draft status before, but I think it has been reopened.

**Comment by anikdhabal** [comment]
It was fine till merge conflict occurred. I have tried many times to fix it but don't know how extra file is added every time.I am done now. Please give some tips to solve it.

**Comment by anikdhabal** [comment]
It was fine before conflict. Then problem occurred after fixing it. I have tried many times fix it but failed. Please suggest some steps to fix it.

**Comment by anikdhabal** [comment]
Today I've learned a lot of git commands and finally fix it😂. Waiting for Review.

**Comment by PeerRich** [comment]
react-beautiful-dnd not maintained anymore. closing in favor of https://github.com/calcom/cal.com/pull/10608 who was also earlier

**Comment by anikdhabal** [comment]
> react-beautiful-dnd not really maintained anymore

I think there isn't another library that is as maintained and an active community.

**Comment by anikdhabal** [comment]
I've added drag-and-drop support to the event list without using any external libraries, and it's working fine. Previously, when we were using the React dnd library, autoAnimate was neglected. However, now auto-animate is applied both to buttons and drag-and-drop. Waiting for review @PeerRich

**Comment by Jaibles** [comment]
Hi @anikdhabal, thank you for you work on this. While it works great, the preference for the interaction on an event type list item is definitely first and foremost, to open it to view. It's nice to be able to re-order, but it's something our users do very seldomly, so we don't want to make it the primary interaction on the event type list item. Closing for now.

**Comment by anikdhabal** [comment]
> 

However, it's an additional feature alongside the button. It's particularly useful when dealing with a large list size. I invested a lot of time in refining it, but in the end, it was closed😓.

**Comment by anikdhabal** [comment]
It's additional feature with the button. The primary interaction should be button as default. It's very useful when the list size is large @PeerRich . The issue is still open, but my PR was closed🙂. I spent a good amount of time on it.

**Pull request closed without merging**